Restructure and rename to make this repo consistent with other RockLib repos

JustinMoss opened the pull request.

JustinMoss committed:
Restructure and rename to make this repo consistent with other RockLib
repos

bfriesen commented on RockLib.EmbeddedNativeLibrary/RockLib.EmbeddedNativeLibrary.csproj:
```diff
@@ -0,0 +1,30 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;netstandard1.6;net45;</TargetFrameworks>
```
For .net framework, let's target 4.0 and 4.5.1

JustinMoss replied:
This causes an issue due to GetTypeInfo() and IReadOnlyCollection not being available before 4.5.  To change this I would need to add constant bases conditionals in at least a few places.

bfriesen commented on RockLib.EmbeddedNativeLibrary/RockLib.EmbeddedNativeLibrary.csproj:
```diff
@@ -0,0 +1,30 @@
+
+  <PropertyGroup>
+    <PackageId>RockLib.EmbeddedNativeLibrary</PackageId>
+    <PackageVersion>1.1.2</PackageVersion>
```
Should make the version the same as the current tag (2.0.2).

JustinMoss replied:
👍

bfriesen commented on RockLib.EmbeddedNativeLibrary/RockLib.EmbeddedNativeLibrary.csproj:
```diff
@@ -0,0 +1,30 @@
+    <PackageReleaseNotes></PackageReleaseNotes>
+    <Copyright>Copyright 2017 (c) Quicken Loans Corporation. All rights reserved.</Copyright>
+    <PackageTags>rocklib embedded native library</PackageTags>
+    <Version>1.1.0</Version>
```
Can we change this to 2.0.0?

JustinMoss replied:
👍

bfriesen commented on RockLib.EmbeddedNativeLibrary/RockLib.EmbeddedNativeLibrary.csproj:
```diff
@@ -0,0 +1,30 @@
+    <Description>Consume native libraries from .NET by adding as embedded resources.</Description>
+    <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
+    <PackageReleaseNotes></PackageReleaseNotes>
+    <Copyright>Copyright 2017 (c) Quicken Loans Corporation. All rights reserved.</Copyright>
```
Copyright years should be 2015-2018.

JustinMoss replied:
👍

bfriesen reviewed (changes requested):
A few things in the csproj.

JustinMoss committed:
Updates from code review

Update csproj package and assembly version
Setup constants for package vs shared versions
Change to c# 3 to allow most amount of support
Change frameworks to netstandard2.0;net40,net451

bfriesen reviewed (approved):
👍
